### ISSUE TITLE
It seems there is a logical error in synthetic_dataset.py

### ISSUE THREAD

**Issue by hust-cxl** [vivjay30/Cone-of-Silence#12]
In synthetic_dataset.py get_mixture_and_gt() funtion, about in line 161
![image](https://user-images.githubusercontent.com/30313363/116178931-f7189780-a748-11eb-9fef-ee2193541226.png)

shifted_gt, _ = utils.shift_mixture(np.stack(gt_waveforms), target_pos, self.mic_radius, self.sr)
should be in the outside of the loop like this:
![image](https://user-images.githubusercontent.com/30313363/116179007-18798380-a749-11eb-8ffa-3e997eee61b1.png)

**Comment by zhangshengoo** [comment]
I get overfitting results during training. The training loss is decreasing but the test loss is almost unchanged. I adopt the same model parameters and the same size of dataset mentioned in the article. Have you encountered this problem？Sorry to bother you.

**Comment by hust-cxl** [comment]
> I get overfitting results during training. The training loss is decreasing but the test loss is almost unchanged. I adopt the same model parameters and the same size of dataset mentioned in the article. Have you encountered this problem？Sorry to bother you.

I don't know whether your test set is "real test set",  I mean if you did not change any code of this project, the log output like "Done with training, going to testing, Test set: Average Loss: XXXX" you see is not real test set loss, it is validation loss instead. It depends on your data division: 
i) data set for validation was generated from your training data(part of your training set)
ii) data in your validation set has not been seen in training set
if your data division was same as i), maybe you ara facing overfitting, but if your stragety was same as ii), maybe you should consider the training data set itself.

**Comment by hust-cxl** [comment]
> I get overfitting results during training. The training loss is decreasing but the test loss is almost unchanged. I adopt the same model parameters and the same size of dataset mentioned in the article. Have you encountered this problem？Sorry to bother you.

I don't know whether your test set is "real test set", I mean if you did not change any code of this project, the log output like "Done with training, going to testing, Test set: Average Loss: XXXX" you see is not real test set loss, it is validation loss instead. It depends on your data division:
i) data set for validation was generated from your training data(part of your training set)
ii) data in your validation set has not been seen in training set
if your data division was same as i), maybe you ara facing overfitting, but if your strategy was same as ii), maybe you should consider the training set itself.

**Comment by vivjay30** [comment]
> In synthetic_dataset.py get_mixture_and_gt() funtion, about in line 161
> ![image](https://user-images.githubusercontent.com/30313363/116178931-f7189780-a748-11eb-9fef-ee2193541226.png)
> 
> shifted_gt, _ = utils.shift_mixture(np.stack(gt_waveforms), target_pos, self.mic_radius, self.sr)
> should be in the outside of the loop like this:
> ![image](https://user-images.githubusercontent.com/30313363/116179007-18798380-a749-11eb-8ffa-3e997eee61b1.png)

Thanks for the fix! Indeed this should only be calculated once at the end

**Comment by vivjay30** [comment]
Regarding the overfitting problem: If you are training with purely synthetically rendered data, then you can generate arbitrarily many samples. Consider grabbing more types of noises, or even using other voice examples like Librispeech etc. When training with real data, it is time consuming to gather this data. Therefore it is very easy to overfit. However it is important to train with real data because a network trained on purely synthetic data will not generalize to real data
